Cut 2.20.0-rc.0

beorn7 opened the pull request:
There are a bunch of bug fixes in flight that I would like to see in
if possible. Thus, I'm postponing the RC by one day, but I wanted to
get some early reviews of the CHANGELOG already.

beorn7 commented:
Adding some reviewers that I felt were particularly involved in this release.

Also, @RichiH , FYI so that you know what's coming.

roidelapluie commented on CHANGELOG.md:
```diff
@@ -1,7 +1,35 @@
+* [CHANGE] TSDB: WAL compression is enabled by default. #7410
+* [FEATURE] PromQL: Added `group()` aggregator. #7480
+* [FEATURE] SD: Added Docker Swarm SD. #7420
+* [FEATURE] SD: Added Digital Ocean SD. #7407
```
```suggestion
* [FEATURE] SD: Added DigitalOcean SD. #7407
```

roidelapluie replied:
That is not fixed

beorn7 replied:
Oops. Will fix in next push.

roidelapluie commented on CHANGELOG.md:
```diff
@@ -1,7 +1,35 @@
+* [FEATURE] SD: Added Digital Ocean SD. #7407
+* [FEATURE] SD: Added Openstack config option to query alternative endpoints. #7494
+* [ENHANCEMENT] Configuration: Exit early on invalid config file and signal it with exit code 2. #7399
+* [ENHANCEMENT] Documentation: Added Docker Swarm configuration example. #7542
```
That is part of the new SD feature

beorn7 replied:
Of course, will remove.

roidelapluie commented on CHANGELOG.md:
```diff
@@ -1,7 +1,35 @@
+* [ENHANCEMENT] UI: Improved scrolling when following hash-fragment links. #7456
+* [ENHANCEMENT] UI: React UI renders numbers in alerts in a more human-readable way. #7426
+* [BUGFIX] API: Fixed error status code in the query API. #7435 
+* [BUGFIX] PromQL: Fixed `avg_over_time` for NaN and float64 overflows. #7346
```
```suggestion
* [BUGFIX] PromQL: Fixed `avg` and `avg_over_time` for NaN, Inf, and float64 overflows. #7346
```

beorn7 commented:
Thanks @roidelapluie . All suggestion have been applied, but GH has a problem right now, so they don't show up yet.

brian-brazil commented on CHANGELOG.md:
```diff
@@ -1,7 +1,34 @@
+* [ENHANCEMENT] SD: Reduce timeouts for Openstack SD. #7507
+* [ENHANCEMENT] SD: Use 10m timeout for Consul watches. #7423
+* [ENHANCEMENT] SD: Added AMI meta label for EC2 SD. #7386
+* [ENHANCEMENT] TSDB: Log the start of creating a checkpoint. #7581
```
Is this worth mentioning? Minor log changes aren't usually mentioned.

beorn7 replied:
I thought it's useful to know for those that need this log line. But I don't feel strongly. I'll remove this line unless someone speaks up in favor of keeping it very soon.

beorn7 replied:
Removed!

roidelapluie commented:
That github UI not up to date is annoying.

beorn7 commented:
Added #7342 and #7323, found by @brian-brazil (thanks for the catch, they were so close to the 2.19 release that I missed them in the `git log` stream).

Also added recently merged #7573. 

I'd really like to get #7562 in for this release. It seems good to go but needs final approval by a maintainer.

brian-brazil commented:
:+1:

beorn7 commented:
OK, this should be ready now. I'll wait for at least one final :+1: .

brian-brazil commented:
:+1:

beorn7 commented:
Just to be super sure, I'll wait for all tests to pass.

roidelapluie commented:
/prombench v2.19.2

codesome commented:
Memory RSS is less for this release, I wonder where that is coming from. 
Also, the query timings look weird, as if some part of inner_eval time was shifted to prepare_time in 2.20, but overall the query times seem to be less if you add all of them up (could it be any of #7453 #7503 #7448?).

roidelapluie commented:
The CPU usage is also less important.

codesome commented:
The low watermark is also very low compared to 2.19. I think that is https://github.com/prometheus/prometheus/pull/7562 for the reduction in CPU and the number of watermarks (hence commits to TSDB)?

codesome commented:
That can explain memory usage reduction too

roidelapluie commented:
I have executed prombench on that #7562 PR and I did not notice that.

beorn7 commented:
Perhaps somehow, you benchmarked the wrong diff? I mean, the watermark increase _has_ to approximately half with your change, and that's what we are seeing now.

roidelapluie commented:
That was seen on that prombench too.

roidelapluie commented:
I mean: I did notice the watermark change but did not notice the CPU/Memory changes.

beorn7 commented:
OK, cool.

Then no red flags so far, on the contrary.

roidelapluie commented:
@beorn7 Before cancelling the benchmark, please take and save a 5 minute CPU profile on both instances on high load, so we have a chance to understand what is going on, WDYT?

roidelapluie commented:
high load means once the part of the cycle where all the targets expose many data

roidelapluie commented:
As we are running for 3 days I have ran pprof ; the winner is.... :partying_face:  @pracucci :tada: with #7390 & #7413

[pprof.zip](https://github.com/prometheus/prometheus/files/4944576/pprof.zip)

roidelapluie commented:
![m1](https://user-images.githubusercontent.com/291750/87884667-2f4bb680-ca10-11ea-9a65-aae5248b4777.png)
![m2](https://user-images.githubusercontent.com/291750/87884669-2fe44d00-ca10-11ea-915f-224a20520aed.png)

Thanks @pracucci

beorn7 commented:
@roidelapluie I assume you have taken the pprof profiles you wanted. I'll just cancel the benchmark now. Will then cut rc.1 with the (few) changes so far. Just to play safe…

beorn7 commented:
/prombench cancel

bwplotka commented:
:tada: Awesome work all!

roidelapluie commented:
Yes we have an explanation about the CPU difference. The fact that it is on the read path also explains why the difference can also be seen when the CPU usage is lower.
